examples/plot_model_agnostic_importance.py

```diff
@@ -65,17 +65,9 @@
 # test (:math:`H_0: X_j \perp\!\!\!\perp y | X_{-j}`) for each variable. However,
 # this test is based on a linear model (LogisticRegression) and fails to reject the null
 # in the presence of non-linear relationships.
-selection_features, X_residual, sigma2, y_res = dcrt_zero(
-    X, y, problem_type="classification", screening=False
-)
-_, pval_dcrt, _ = dcrt_pvalue(
-    selection_features=selection_features,
-    X_res=X_residual,
-    y_res=y_res,
-    sigma2=sigma2,
-    fdr=0.05,
-)
-
+d0crt = D0CRT(problem_type="classification", screening=False)
+d0crt.fit(X, y)
+_, pval_dcrt = d0crt.importance(fpr=0.05)
```

Comment:
I think it is already mentioned in another PR, but `importance` should not take and `fpr` argument.

Reply:
No, this was mentioned in issue #217.
However, there is no consensus if this function has a X_test and Y_test or not.

Reply:
all importance methods should have should have an X=None, Y=None, as arguments, that would default to the already provided training data. In some cases, these arguments would not even be taken into account.

Reply:
The fact that all the importance methods will have an X and y is not yet defined. 
Using an X_test and y_test will change a bit the algorithm for computing the residual on this test data.

As a user, it's more readable to add additional optional parameters to a function than to have parameters which don't have any effect.

Reply:
The point is about API uniformity: you want to be able to loop over methods with maximally similar arguments without triggering an error. The mechanism is fundamental in sklearn and related libraries.

Reply:
Can you indicate to me one function in sklearn where some parameters are not used by the function due to the homogenisation of the API?